### PR TITLE
Move record functionalities into robot

### PR DESCRIPTION
**Pull request by nqyy**

### CODE REVIEW

File: mbodied/robots/robot.py [hunk 1]
```diff
@@ -15,21 +15,40 @@
...
-        calculate_action: Calculates the action between two robot states, optional for robot recorder.
+    This class serves as a blueprint for creating interfaces to control robot hardware
+    or other devices. It provides essential methods and guidelines to implement
+    functionalities such as executing motions, capturing observations, and recording
+    robot data, which can be used for training models.
+
+    Key Features:
+    - **Motion Execution:** Implement the `do` method to execute robot movements.
+    - **Data Fetching:** Implement `fetch` to retrieve data from the hardware.
+    - **Continuous Data Capture:** Implement `capture` for continuous data capture.
```
Comment (sebbyjp) [inline mbodied/robots/robot.py:33]: Key feature:
Its not a blueprint or essential methods.

-organize your hardware interface in a modular fashion
-support asynchronous dataset creation


Why not just name methods observer(), get_state(), prepare_action() (or post-process motion()) and do() ?

By the way you shouldn't need to calculate_action() thats the whole point of defining a Motion.

File: examples/5_teach_robot_record_dataset.py [hunk 2]
```diff
@@ -39,11 +36,7 @@
 @click.option("--task", default="pick up the remote", help="The task to perform/record")
 @click.option("--backend", default="openai", help="The backend to use", type=click.Choice(["anthropic", "openai"]))
 @click.option("--backend_api_key", default=None, help="The API key for the backend, i.e. OpenAI, Anthropic")
-@click.option("--disable_audio", default=False, help="Disable audio input/output")
-def main(task: str, backend: str, backend_api_key: str, disable_audio: bool) -> None:
-    if disable_audio:
-        os.environ["NO_AUDIO"] = "1"
-
+def main(task: str, backend: str, backend_api_key: str) -> None:
     context = [
```
Comment (sebbyjp) [inline examples/5_teach_robot_record_dataset.py:40]: agent = LanguageAgent("model_src"="openai", context="Respond in the following json schema {...}")

File: examples/5_teach_robot_record_dataset.py [hunk 3]
```diff
@@ -58,22 +51,34 @@ def main(task: str, backend: str, backend_api_key: str, disable_audio: bool) ->
         model_src=backend,
     )
 
-    # Prefer to use use_pyaudio=False for MAC.
-    audio = AudioAgent(use_pyaudio=False)
-
     # Use sim robot for this example.
```
Comment (sebbyjp) [inline examples/5_teach_robot_record_dataset.py:54]: delete redundant comment

Comment (sebbyjp) [inline examples/5_teach_robot_record_dataset.py:54]: delete this comment i mean

File: examples/5_teach_robot_record_dataset.py [hunk 4]
```diff
@@ -58,22 +51,34 @@ def main(task: str, backend: str, backend_api_key: str, disable_audio: bool) ->
...
     robot = SimRobot()
-    robot_recorder = RobotRecorder(robot, frequency_hz=5)
+    # Initialize the recorder for the robot.
+    robot.init_recorder(
```
Comment (sebbyjp) [inline examples/5_teach_robot_record_dataset.py:57]: with robot.record():


Talk about whats going on under the hood in a docstring or a markdown doc

**Review by sebbyjp — CHANGES_REQUESTED** [review]

File: mbodied/robots/robot.py (hunk 1 above)
Comment (nqyy) [reply mbodied/robots/robot.py]: i renamed them to get_observation(), get_state() and prepare_action()

**Review by sebbyjp — APPROVED** [review]

**Pull request merged**